spring-cloud-task-jdbchdfs-common/src/main/java/org/springframework/cloud/task/jdbchdfs/common/HdfsTextItemWriter.java

```diff
@@ -0,0 +1,232 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.springframework.cloud.task.jdbchdfs.common;
+
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.CommonConfigurationKeysPublic;
+import org.apache.hadoop.fs.Path;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.springframework.batch.item.ExecutionContext;
+import org.springframework.batch.item.WriteFailedException;
+import org.springframework.batch.item.file.transform.LineAggregator;
+import org.springframework.batch.item.support.AbstractItemStreamItemWriter;
+import org.springframework.beans.factory.InitializingBean;
+import org.springframework.data.hadoop.store.DataStoreWriter;
+import org.springframework.data.hadoop.store.output.OutputStreamWriter;
+import org.springframework.data.hadoop.store.strategy.naming.ChainedFileNamingStrategy;
+import org.springframework.data.hadoop.store.strategy.naming.FileNamingStrategy;
+import org.springframework.data.hadoop.store.strategy.naming.RollingFileNamingStrategy;
+import org.springframework.data.hadoop.store.strategy.naming.StaticFileNamingStrategy;
+import org.springframework.data.hadoop.store.strategy.rollover.RolloverStrategy;
+import org.springframework.data.hadoop.store.strategy.rollover.SizeRolloverStrategy;
+import org.springframework.util.Assert;
+import org.springframework.util.StringUtils;
+
+/**
+ * Writes items as a byte array to a destination as specified by the fsURI.
+ *
```

Comment:
We should note that this `ItemWriter` implementation does not manage the state of the file like the `FlatFileItemWriter` does.  I'm assuming that's because there isn't a way to truncate bad data at the end of a file on HDFS.